### ISSUE TITLE
role="listbox" attribute missing from <ul> within terra-list component

### ISSUE THREAD

**Issue by juliacalandro** [cerner/terra-core#1656]
# Bug Report

## Description
When we upgraded to version 2.7.0 of terra-list, we received the following accessibility violation from our wdio tests:

_"description": "Ensures elements with an ARIA role that require parent roles are contained by them",
  "help": "Certain ARIA roles must be contained by particular parents",
  "helpUrl": "https://dequeuniversity.com/rules/axe/2.6/aria-required-parent?application=axeAPI",
  "id": "aria-required-parent",
..................
"nodes": [
    {
      "all": [],
      "any": [
        {
          "data": [
            "listbox"
          ],
          "id": "aria-required-parent",
          "impact": "critical",
          "message": "Required ARIA parent role not present: listbox",
          "relatedNodes": []
        }
      ],
      "failureSummary": "Fix any of the following:\n  Required ARIA parent role not present: listbox"_

After inspecting the terra-list code, we noticed that the `role="option"` attribute was getting set on `<li>` within the ListItem component, but a role attribute was not getting set on `<ul>` within the List component. I looked into this issue a bit further, and found [this webpage](http://w3c.github.io/aria-practices/examples/listbox/listbox-scrollable.html) about aria/accessibility which states that `role="listbox"` needs to be an attribute on `<ul>` which is the parent of `<li>`.

## Steps to Reproduce
Install the aXe extension for Chrome.
Run a project using terra-list so that it can be opened and inspected in the browser.
Inspect the page and navigate to where the List component is being used.
Verify that `<ul>` does not have a role set but its child `<li>`s do.
Navigate to the aXe tab and run an inspection on the page.
Verify that an accessibility error is thrown with a description matching the error description above.
Add `role="listbox"` as an attribute to each offending `<ul>` component and run the aXe inspection again.
Verify that the accessibility issue doesn't exist anymore.

## Additional Context / Screenshots
<img width="684" alt="screen shot 2018-06-27 at 11 22 05 am" src="https://user-images.githubusercontent.com/39311706/41986676-6e051fc6-79fc-11e8-86b7-47038b36182f.png">

## Expected Behavior
I expect the accessibility issue mentioned above to no longer be thrown by the terra-list component.

## Possible Solution
Add missing `role="listbox"` attribute to `<ul>` within terra-list component.

## Environment
terra-list 2.6.0
Google Chrome Version 67.0.3396.99 (Official Build) (64-bit)

**Comment by bjankord** [comment]
I'm unable to reproduce this in the latest version of [terra-list (v2.15.0)](https://github.com/cerner/terra-core/blob/master/packages/terra-list/package.json#L4)

Can your team upgrade to that version?

**Comment by juliacalandro** [comment]
Yes we get the same error in every version greater than 2.6.0. This problem is being pulled in through the terra-clinical-item-collection component which utilizes terra-list.

**Comment by bjankord** [comment]
I was able to reproduce the issue on the [terra-clinical test page for selectable item-collections](https://engineering.cerner.com/terra-clinical/#/tests/terra-clinical-item-collection/clinical-item-collection/item-collection-selectable). I'm going to close this issue and log one to the terra-clinical repo.

We need to update the component to add `role="listbox"` to [this component](https://github.com/cerner/terra-clinical/blob/master/packages/terra-clinical-item-collection/src/_ListView.jsx#L45) if the item collection component is selectable.
